termqt/terminal_widget.py

```diff
@@ -210,23 +264,28 @@ def _paint_buffer(self):
             if real_ln < 0 or real_ln >= len(self._buffer):
                 break
 
-            row = self._buffer[ln + offset]
+            row = self._buffer[real_ln]
 
             ht += lh
             for cn, c in enumerate(row):
                 if c:
+                    is_selected = self._is_selected(cn, real_ln)
```

Comment:
For the sake of executing speed, I would be reluctant to place a method call inside the paint loop. Given that the selection region is sequential, I think this part of the code can be optimized and put inside the loop.